test_runner: add --test-skip-pattern cli option (#52529)
nodejs/node: doc/api/cli.md

whose name matches the provided pattern. See the documentation on
[filtering tests by name][] for more details.

> If both `--test-name-pattern` and `--test-skip-pattern` are supplied,

Reviewer: This statement is used a few times throughout this PR, but it is a little vague. How about something along the lines of:

> If a test name matches both `--test-name-pattern` and `--test-skip-pattern`, the `--test-name-pattern` flag takes precedence.

> If both the `--test-name-pattern` and `--test-skip-pattern` arguments are

Reviewer: I still think this is vague. This could be interpreted as "if you pass both flags, `--test-skip-pattern` is ignored globall." That's why I recommended text that was specific to individual tests.

Author: Well, it's not wrong. If `--test-name-pattern` is supplied, `--test-skip-pattern` *is* ignored globally. In the checking algorithm, it checks the patterns like:
```js
    if (testNamePatterns) {
      return !testMatchesPattern(this, testNamePatterns);
    }

    if (testSkipPatterns) {
      return testMatchesPattern(this, testSkipPatterns);
    }
```

Author: I could change it to, but I'd need to test to see if it'd work differently:
```js
return !testMatchesPattern(this, testNamePatterns) && testMatchesPattern(this. testSkipPatterns);
```

Reviewer: Fair enough. I'm OK with this text.

Author: Don't worry, I changed the logic a bit, and now, if they are both supplied, any tests need to match **both** arguments. I think that makes more sense.

> `--test-name-pattern` will take precedence

Reviewer 2: ```suggestion
`--test-name-pattern` will take precedence.
```

node --test --test-shard=3/3
```

### `--test-skip-pattern`
<!-- YAML
added:
  - REPLACEME
-->

A regular expression that configures the test runner to only execute tests

> whose name don't match the provided pattern. See the documentation on

Reviewer 2: ```suggestion
whose name doesn't match the provided pattern. See the documentation on
```

[filtering tests by name][] for more details.

If both `--test-name-pattern` and `--test-skip-pattern` are supplied,

> `--test-name-pattern` will take precedence

Reviewer 2: ```suggestion
`--test-name-pattern` will take precedence.
```

node --test --test-shard=3/3
```

### `--test-skip-pattern`

<!-- YAML
added:
  - REPLACEME
-->

> A regular expression that configures the test runner to only execute tests
> whose name doesn't match the provided pattern. See the documentation on

Reviewer: ```suggestion
A regular expression that configures the test runner to skip tests
whose name matches the provided pattern. See the documentation on
```
